File: src/problem010.py
def euler_010(upper_bound):
    sieve = [False, False, True] + [True] * (upper_bound - 3)
    partial_sum = 0
    for n, p in enumerate(sieve):
        if p:
            partial_sum += n
            for k in range(n * 2, upper_bound, n):
                sieve[k] = False
    return partial_sum

if __name__ == '__main__':
    # print(euler_010(10))
    print(euler_010(2000000))
            

File: src/go/problem_010.go
package main

import (
	"fmt"
)

const (
	upper_bound = 2000000
)

var (
	sieve = make([]bool, upper_bound)
)

func prepareSieve() {
	sieve[2] = true
	for i := 3; i < upper_bound; i += 2 {
		sieve[i] = true
	}

	for i := 3; i < upper_bound; i += 2 {
		if sieve[i] {
			for k := i * 2; k < upper_bound; k += i {
				sieve[k] = false
			}
		}
	}
}

func main() {
	prepareSieve()

	// sum primes
	var primeSum int64 = 0
	for i, isPrime := range sieve {
		if isPrime {
			primeSum += int64(i)
		}
	}

	fmt.Println(primeSum)
}